src/components/region-select.vue

```diff
@@ -1,70 +1,70 @@
+  import regions from 'country-region-data'
+  export default {
+    name: 'RegionSelect',
+    props: ['country', 'region', 'defaultRegion', 'countryName', 'regionName', 'class'],
+    data: () => ({
+      shownRegions: [],
+      regions,
+    }),
+    mounted() {
+      if (this.country) {
+        this.getRegionWithCountry()
       } else {
-        findRegion = this.defaultRegion ? this.defaultRegion : 'US'
-      }
-      const regionObject = regions.find((elem) => {
+        let findRegion = ''
         if (this.countryName) {
-          return elem.countryName === findRegion
+          findRegion = this.defaultRegion ? this.defaultRegion : 'United States'
         } else {
-          return elem.countryShortCode === findRegion
+          findRegion = this.defaultRegion ? this.defaultRegion : 'US'
         }
-      })
-      this.shownRegions = regionObject.regions.map((elem) => elem)
-    }
-  },
-  computed: {
-    valueType() {
-      return this.regionName ? 'name' : 'shortCode'
-    }
-  },
-  methods: {
-    onChange(region) {
-      this.$emit('input', region)
+        const regionObject = regions.find((elem) => {
+          if (this.countryName) {
+            return elem.countryName === findRegion
+          } else {
+            return elem.countryShortCode === findRegion
+          }
+        })
+        this.shownRegions = regionObject.regions.map((elem) => elem)
+      }
     },
-    getRegionWithCountry() {
-      const regionObject = regions.find((elem) => {
-        if (this.countryName) {
-          return elem.countryName === this.country
+    computed: {
+      valueType() {
+        return this.regionName ? 'name' : 'shortCode'
+      }
+    },
+    methods: {
+      onChange(region) {
+        this.$emit('input', region)
+      },
+      getRegionWithCountry() {
+        const regionObject = regions.find((elem) => {
+          if (this.countryName) {
+            return elem.countryName === this.country
+          } else {
+            return elem.countryShortCode === this.country
+          }
+        })
+        this.shownRegions = regionObject.regions.map((elem) => elem)
+      }
+    },
+    watch: {
+      country(newVal, oldVal) {
+        if (oldVal !== '') {
+          this.onChange('')
+        }
+        if (this.country) {
+          this.getRegionWithCountry()
         } else {
-          return elem.countryShortCode === this.country
+          this.shownRegions = []
         }
-      })
-      this.shownRegions = regionObject.regions.map((elem) => elem)
-    }
-  },
-  watch: {
-    country(newVal, oldVal) {
-      if (oldVal !== '') {
-        this.onChange('')
-      }
-      if (this.country) {
-        this.getRegionWithCountry()
-      } else {
-        this.shownRegions = []
       }
     }
   }
-}
 </script>
 
 <template>
-  <select @change="onChange($event.target.value)">
+  <select @change="onChange($event.target.value)" class="class">
```

Comment:
please change class to :class="className". You need the : to achieve what you are going for here.